text/dependencies/rfcs/0000-dependency-management-phase-one.md

```diff
@@ -0,0 +1,287 @@
+
+Since metadata and known versions won't be stored data anymore, buckets will
+only be needed to store dependencies that have been compiled or processed in
+some way. For these cases, a dependency-specific bucket set up in Paketo
+Dependencies project within GCP to store data. The directory name will be the
+name of the dependency and it will contain dependency archives (if compiled).
+Push credentials to this bucket will be reserved for the buildpack and
+maintainers, but the contents will be publicly available.
+
+### Remove Known Versions
+
+As mentioned above, dependency workflows will use the `buildpack.toml` as a
+source of the latest versions we support. Because of this change, we will no
+longer need to keep track of all known versions in a separate file.
+
+### Transfer Version Retrieval Code
+
+Version discovery code comes from
+[dep-server/pkg/dependency](https://github.com/paketo-buildpacks/dep-server/tree/main/pkg/dependency).
+Each dependency has a slightly different way new versions are made
+discoverable. This code should be ported from the dep-server repository to the
+buildpack under a directory named `dependency/retrieval`. It should be
+well-documented and useable on a local environment. The location must be
+standardized for use in automation.
+
+### Transfer Metadata Generation Code
+Metadata generation code also comes from
+[dep-server/pkg/dependency](https://github.com/paketo-buildpacks/dep-server/tree/main/pkg/dependency).
+Code to generate/gather all of the metadata for a dependency should be moved
+into the buildpack under the `dependency/metadata` directory.  The
+code should do essentially the same thing that the existent code does, and
+support the same fields.
+
+The code for getting the `source URI`, `version`, `URI`, `SHA256`, `ReleaseDate`,
+`DeprecationDate`, and `CPE` fields are all dependency-specific and can live in
+the buildpack `dependency/metadata` location.
+
+The `PURL` and `licenses` fields are more generic across dependencies, so the
+code for generating them should come from a common location, and used as
+library in the dependency-specific metadata code to reduce code duplication.
+The dep-server repository will retain this code, which can be imported as a
+libray in the dependency-specific metadata code can use it as a library.
+
+#### Caveat: Compiled Dependencies
+In the case that the dependencies need to be compiled or processed, the
+metadata generation code should omit the `URI` and the `SHA256` from the
+metadata. This will be used in automation (described in detail in a subsequent
+RFC) to let the dependency management system to trigger compilation of the
+dependency. When the dependency is compiled and uploaded to a bucket, the
+bucket URI will be the URI in the metadata, and the compiled dependency SHA256
+will be the SHA256 in the metadata.
+
+### Smoke Test
+In the dep-server, a smoke test is run against every dependency before the
```

Comment:
I haven't fully visualized the order of operations, so this might be in the wrong place, but I think it's valuable to run some sort of test before bumping the dependency in the source of truth in all cases - not just compiled dependencies.

We should definitely test compiled/processed dependencies before uploading them (and hence making them the source of truth), but I'm also thinking about the situation where a dependency that we take directly from upstream results in broken behavior.

I think it's good enough for now to have a PR model (which runs the acceptance tests) for a potential version bump. That's basically what we have now - i just don't want to lose that feature.

Reply:
I would recommend not reading these until they're undrafted, theyre undergoing changes.